frontend/src/components/common/Notifications/index.tsx

```diff
@@ -0,0 +1,151 @@
+    '&.MuiMenuItem-root': {
+      borderBottom: `1px solid ${theme.palette.notificationBorderColor}`,
+      padding: '1rem',
+    },
+  },
+  notificationMessage: {
+    '&': {
+      wordBreak: 'break-word',
+      whiteSpace: 'normal',
+    },
+  },
+  root: {
+    '& .MuiPaper-root': {
+      width: '30vw',
+      minWidth: '300px',
+      maxHeight: '400px',
+    },
+  },
+  errorItem: {
+    '&': {
+      color: theme.palette.error,
+    },
+  },
+}));
+
+function prepareNotificationsList(
+  notifications: KubeEvent[] | undefined,
+  apiError: ApiError | undefined,
+  clickEventHandler: (event?: any) => void
+) {
+  const classes = useStyles();
+  if (!notifications || notifications.length === 0) {
+    return (
+      <MenuItem onClick={clickEventHandler} className={classes.notificationItem}>
+        You don't have any notifications right now
+      </MenuItem>
+    );
+  }
+
+  return !apiError ? (
+    notifications?.map((notification, index: number) => {
+      return (
+        <MenuItem
+          onClick={clickEventHandler}
+          key={`${notification}__${index}`}
+          className={classes.notificationItem}
+        >
+          <Grid container justifyContent="flex-end" spacing={2}>
+            <Grid item md={12}>
+              {notification.message}
+            </Grid>
+            <Grid item>{timeAgo(new Date(notification.metadata.creationTimestamp))}</Grid>
+          </Grid>
+        </MenuItem>
+      );
+    })
+  ) : (
+    <MenuItem onClick={clickEventHandler} className={classes.errorItem}>
+      {apiError.message}
+    </MenuItem>
+  );
+}
+
+export default function Notifications() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const selectedClusterName = getCluster();
+  const dispatch = useDispatch();
+  const clusters = useTypedSelector(state => state.config.clusters);
+  const history = useHistory();
+  let notificationsAlert;
+  let notifications;
+  let apiError: ApiError | undefined;
+
+  if (selectedClusterName && clusters) {
+    apiError = clusters[selectedClusterName].apiError;
+    notifications = clusters[selectedClusterName].notifications;
+    notificationsAlert = clusters[selectedClusterName].notificationAlert;
+  }
+
+  const handleClick = (event: any) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+    if (selectedClusterName) {
+      dispatch(setClusterNotificationAlert(selectedClusterName, false));
+    }
+  };
+
+  function menuItemClickHandler() {
+    if (selectedClusterName) {
+      dispatch(setClusterNotificationAlert(selectedClusterName, false));
+    }
+    setAnchorEl(null);
+    //only take us to events page if it's not a error notification from fetching events
+    if (!apiError) {
+      history.replace('/');
+    }
+  }
+
+  return (
+    <>
+      <IconButton onClick={handleClick}>
+        {notifications && notificationsAlert && notifications.length !== 0 ? (
```

Comment:
The badge should be set on the notifications based on whether we have "unread" notifications, and this should be done without having to check a redux status, in order for the functionality to be encapsulated and handled within the component.

Reply:
Fixed